SmartDeviceLink/SDLH264ByteStreamPacketizer.m

```diff
@@ -0,0 +1,42 @@
+//
+//  SDLH264ByteStreamPacketizer.m
+//  SmartDeviceLink-iOS
+//
+//  Created by Sho Amano on 4/11/17.
+//  Copyright © 2017 Xevo Inc. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "SDLH264ByteStreamPacketizer.h"
+
+@interface SDLH264ByteStreamPacketizer () <SDLH264Packetizer>
+@property (nonatomic) NSData *startCode;
+@end
+
+@implementation SDLH264ByteStreamPacketizer
+
+- (instancetype)init {
+    if (self = [super init]) {
+        // This is the start code that we will write to the elementary stream before every NAL unit
+        UInt8 startCode[] = {0x00, 0x00, 0x00, 0x01};
+        _startCode = [[NSData alloc] initWithBytes:startCode length:4];
+    }
+
+    return self;
+}
+
+- (NSArray *)createPackets:(NSArray *)nalUnits pts:(double)pts {
+
+    NSMutableArray *array = [NSMutableArray arrayWithCapacity:1];
```

Comment:
This is because the previous implementation consolidated all data into one NSArray, and I didn't want to change the behavior. I put a comment in commit https://github.com/smartdevicelink/sdl_ios/pull/703/commits/107382ff40ed37c794096cd9740d0ce079b01a06 for clarification.

I'm OK to change the behavior, so please let me know if I should go ahead to change.

Reply:
The capacity is just a rough estimate to allow the array to not need to extend itself (which occurs when it adds an additional item over capacity), so if we know how many are in there, having an exact capacity (or slightly larger) is preferable to slightly help performance.

Reply:
Yes I agree.
I think I should have been more clear. Current implementation consolidates all NAL units into one NSData buffer then forward it to delegate. Please refer to https://github.com/smartdevicelink/sdl_ios/blob/release/5.0.0/SmartDeviceLink/SDLVideoEncoder.m#L164.
Since I hesitated to change this behavior, I also created one NSMutableData here, appended all NAL units into it and added it to NSMutableArray. So that is why capacity of the array is always 1.